File: VB/B186127/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraScheduler

Namespace Sample
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private day As Integer = 0
		Private privateSchedulerStorage As SchedulerStorage
		Public Property SchedulerStorage() As SchedulerStorage
			Get
				Return privateSchedulerStorage
			End Get
			Set(ByVal value As SchedulerStorage)
				privateSchedulerStorage = value
			End Set
		End Property

		Private Sub AptTimeLabel_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles AptTimeLabel.BeforePrint
			Dim apt As Appointment = TryCast(GetCurrentRow(), Appointment)
			Dim label As XRLabel = TryCast(sender, XRLabel)

            If (apt.AllDay) Then
                label.Text = "All day"
            Else
                label.Text = String.Format("{0} - {1}", apt.Start.ToShortTimeString(), apt.End.ToShortTimeString())
            End If
        End Sub

		Private Sub AptDescriptionLabel_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles AptDescriptionLabel.BeforePrint
			Dim apt As Appointment = TryCast(GetCurrentRow(), Appointment)

			Dim label As XRLabel = TryCast(sender, XRLabel)
			label.Text = String.Format("{0} {1}" & Constants.vbLf & "{2}", apt.Subject, apt.Location, apt.Description)
			label.BackColor = SchedulerStorage.GetLabelColor(apt.LabelId)
		End Sub

		Private Sub Detail_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles Detail.BeforePrint
			day = Convert.ToDateTime(GetCurrentColumnValue("Start")).Day
		End Sub

		Private Sub GroupHeader1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles GroupHeader1.BeforePrint
			e.Cancel = day = Convert.ToDateTime(GetCurrentColumnValue("Start")).Day
		End Sub
	End Class
End Namespace

File: CS/B186127/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.XtraScheduler;

namespace Sample {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        int day = 0;
        public SchedulerStorage SchedulerStorage { get; set; }

        private void AptTimeLabel_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            Appointment apt = GetCurrentRow() as Appointment;
            XRLabel label = sender as XRLabel;

            label.Text = apt.AllDay ? "All day" : label.Text = string.Format("{0} - {1}", apt.Start.ToShortTimeString(), apt.End.ToShortTimeString());
        }

        private void AptDescriptionLabel_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            Appointment apt = GetCurrentRow() as Appointment;
            
            XRLabel label = sender as XRLabel;
            label.Text = string.Format("{0} {1}\n{2}", apt.Subject, apt.Location, apt.Description);
            label.BackColor = SchedulerStorage.GetLabelColor(apt.LabelId);
        }
        
        private void Detail_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            day = Convert.ToDateTime(GetCurrentColumnValue("Start")).Day;
        }

        private void GroupHeader1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            e.Cancel = day == Convert.ToDateTime(GetCurrentColumnValue("Start")).Day;
        }
    }
}
